### ISSUE TITLE
Orgs Schema V2 + Looming Questions

### ISSUE THREAD

**Issue by JonEsparaz** [jesus-collective/mobile#127]
Still some questions to be answer, but for now we definitely need the following changes:

- [x] The ability for an organization to create sub-organizations. 
Ex. The Meeting House -> TMH Uptown Toronto, TMH Oakville, etc
- [x] An individual can only be part of **one** organization (we don't need many-to-many connections)

Stuff we don't know yet:
- Can an organization post, create events, etc as the organization (or is it an admin on behalf of the organization - if so what's the difference between an individual's post and an org's post)?
- How are sub-organizations different from organizations?
- Do we actually need to allow for multiple admins (#107)?
- What "special" things can organizations do (ex publish curriculum) and how does this work? How do the JC admins manually control this?
- Since the admin account and the organization are tied together in one payment, what happens if that admin leaves the organization in real life? What happens on Jesus Collective? Do they have to pay again? Leave the platform? Are they just stuck forever?
- When you add a member to your organization, do they have to sign up (and pay) as individuals? Or is it a package deal in terms of payment? What if the new member already have a individual account?
- Can an individual user request to join an org (#104)?

**Comment by JonEsparaz** [comment]
@lucastbelem I think most of what we discussed is here.

**Comment by lucastbelem** [comment]
Some answers for you @JonEsparaz and @GeorgeBellTMH:

- Organizations can post as orgs, but we'd like the ability to have a human's name attached to it. I.e. Lucas Belem posting as The Meeting House. 

- Multiple admins should be allowed, but there should be an owner-admin who has access to billing and maybe some other special features.

- Orgs cannot publish their own resources. Resources must be added by the JC admin team and assigned to an org. They exist very much like a profile, except that they have resources (which they can't create on their own). Still waiting on answers on specials for events, groups, message boards.

- If an admin leaves the company, admin ownership needs to be transferred to someone else on the org. We need to make it possible for this new person to be someone who isn't on the platform yet (i.e. the new hire replacing that person who needs to still create their account).

- Members of org do not need to pay. Orgs pay and that gives them the ability to invite an unlimited number of individuals as members of the org, who themselves do not pay. 

Still need to know:

- Can an organization create events, etc as the organization (or is it an admin on behalf of the organization)?
- What if the new member already have a individual account?
- Can an individual user request to join an org (#104)?

**Comment by GeorgeBellTMH** [comment]
We've designed resources to be created by any user....or if eventually orgs want to do it they can as well...for now we will lock it down to only JC admins...

**Comment by lucastbelem** [comment]
Here are all of the confirmed answers/resolutions to orgs: fyi - many-to-many may be needed after all.


- Organizations and post and appear as the organization. Individual posting as the organization should appear in a secondary way design-wise. i.e. Zulema Smith posting as The Meeting House OR The Meeting House (Zulema Smith).
- Organizations are allowed to have an unlimited number of individual members.
- Organizations can exist on the platform as long as there is at least 1 user attached to it (i.e. the administrator is sufficient for making the organization visible and findable).
- Organizations need to be able to create sub-organizations (i.e. multi site churches).
- THIS IS A CHANGE: Individuals can be members of multiple organizations and get access to the highest membership tier within all organizations they are a member of. For future development: a directory of organization members.
- Organizations can create events and appear as the owner of the event.
- Organizations can invite people outside and inside of the platform, but individuals cannot request to join an organization. If an individual already has a profile on the platform, it is up to the organization to invite that individual to join. 

@JonEsparaz what other answers do you need?

**Comment by JonEsparaz** [comment]
@lucastbelem per the change... How do we then determine which organization appears on the user’s profile?

**Comment by JonEsparaz** [comment]
@lucastbelem how about groups? Can they create groups? Also, can the organization join groups (and I assume it doesn’t make sense to allow organizations to attend events)? 

Also, I believe it was intended that users can follow an organization (then get alerts when they create an event or have new curriculum) - is this actually a thing?

**Comment by GeorgeBellTMH** [comment]
I think on a user profile it should show a list of organizations you are following, and a list of organizations you are a member of.

**Comment by JonEsparaz** [comment]
@lucastbelem two more questions:
- To explicitly  confirm, organizations can only be created via the sign up flow and an admin account **must** be created along with the organization (you cannot attach an existing account) - this will be painful to change later, so it would be good to know for sure now
- Do the contents of the Google Sheet supersede what's on Figma in terms of content on an organization's profile?

**Comment by JonEsparaz** [comment]
> I think on a user profile it should show a list of organizations you are following, and a list of organizations you are a member of.

Ok this makes sense to me. What happens to the information a user input when creating their profile though? (It would look weird if they are a member of the Meeting House, but the information about TMH conflicts with the org's profile)...

**Comment by GeorgeBellTMH** [comment]
I think anyone who is a member of an organization has this information disappear....it's superseded by the organizations information...

**Comment by GeorgeBellTMH** [comment]
If the only time to create an org is at signup then users have to delete themselves to create an org...it should be possible through both flows, unless we want to lock users into not have orgs once they've made that decision...

**Comment by GeorgeBellTMH** [comment]
I actually think we will shortly realize that instead of collecting org information on the user profile, we should just ask what organizations they are apart of...and then handle that after they finish their user profile (so if they say they are apart of an existing org we handle that via an approval process, if they say they are part of a new org, we do that flow...and if they say they are not part of an org - or don't want to create one - we give them the current design...)

**Comment by GeorgeBellTMH** [comment]
I've create an example in Figma...https://www.figma.com/file/aJzmRbxyuoqkRPUikSVwUgwk/Jesus-Collective?node-id=3343%3A148164

**Comment by JonEsparaz** [comment]
This might contradict one of Lucas' earlier points:

> * Organizations can invite people outside and inside of the platform, but individuals cannot request to join an organization. If an individual already has a profile on the platform, it is up to the organization to invite that individual to join.

**Comment by lucastbelem** [comment]
@JonEsparaz 

- Regarding orgs that appear: I like George's idea of displaying all of them. I don't know what this would look like though. Showing the orgs that the user follows should be easy - we just add another row of tiles to their profile like "my events" and "my groups".

- Regarding groups: Orgs can create groups (i.e. The Meeting House would be able to create groups). Orgs cannot join groups.

- Regarding events: Orgs can create events, but cannot attend events.

- Regarding following orgs: You're correct. Users can follow orgs and be notified of new resources, new events, new groups by that org. On that note, this should be added to user settings when we get to notifications (i.e. do you want to be notified when: 1 - the org posts a resource; 2 - the org posts an event; 3 - the org posts a group, etc.)

- Regarding org creation: Yes, they can only be created through the sign up flow and an admin must be connected to it before it exists. I.e. if the user doesn't want to finish creating their admin profile, they can save as draft and finish later, but cannot submit their org profile until they complete their admin one.

- Regarding orgs info: Org info on org profile supersedes org info on individual profile.

- Regarding the google sheet vs Figma: Google sheet takes priority.

**Comment by lucastbelem** [comment]
@GeorgeBellTMH regarding your figma design - let's talk about this. There is a use case in my mind that won't work with some of things we're thinking.

**Comment by GeorgeBellTMH** [comment]
https://www.figma.com/file/aJzmRbxyuoqkRPUikSVwUgwk/Jesus-Collective?node-id=3343%3A148695

**Comment by GeorgeBellTMH** [comment]
That's an example of how the profile page could look

**Comment by lucastbelem** [comment]
> That's an example of how the profile page could look

Love it.

**Comment by JonEsparaz** [comment]
> Regarding orgs info: Org info on org profile supersedes org info on individual profile.

Again for this, if they are in multiple organization, how do we decide what information to use to supersede? Does the user somehow select "make this my primary organization"?

**Comment by lucastbelem** [comment]
Let's talk about stuff in 10 mins at our 1:15pm.

**Comment by lucastbelem** [comment]
You guys wanna meet now?

**Comment by GeorgeBellTMH** [comment]
Sure...

**Comment by JonEsparaz** [comment]
Closing since I have all the answers I need for now. If you need to reference it, check out the [Organizations project](https://github.com/jesus-collective/mobile/projects/5).
